presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxStatsDumper.java

```diff
@@ -0,0 +1,97 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.jmx;
+
+import com.facebook.presto.spi.SchemaTableName;
+import io.airlift.log.Logger;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class JmxStatsDumper
+        implements Runnable
+{
+    private static final Logger log = Logger.get(JmxStatsDumper.class);
+
+    private final JmxStatsHolder jmxStatsHolder;
+    private final JmxMetadata jmxMetadata;
+    private final JmxRecordSetProvider jmxRecordSetProvider;
+    private final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+    private final long period;
+    private long nextDump = -1;
+
+    public JmxStatsDumper(JmxStatsHolder jmxStatsHolder, JmxMetadata jmxMetadata, JmxRecordSetProvider jmxRecordSetProvider, long period)
+    {
+        this.jmxStatsHolder = requireNonNull(jmxStatsHolder, "jmxStatsHolder is null");
+        this.jmxMetadata = requireNonNull(jmxMetadata, "jmxMetadata is null");
+        this.jmxRecordSetProvider = requireNonNull(jmxRecordSetProvider, "jmxRecordSetProvider is null");
+        this.period = requireNonNull(period, "period is null");
+    }
+
+    public void start()
+    {
+        long nowTs = System.currentTimeMillis();
+        nextDump = roundToPeriod(nowTs) + period;
+        schedule(nowTs);
+    }
+
+    private void schedule(long nowTs)
+    {
+        executor.schedule(this, Math.max(nextDump - nowTs, 0), TimeUnit.MILLISECONDS);
+        nextDump += period;
+    }
+
+    private long roundToPeriod(long ts)
+    {
+        return (ts / period) * period;
+    }
+
+    @Override
+    public void run()
+    {
+        long nowTs = System.currentTimeMillis();
+        for (String tableName : jmxStatsHolder.listTables()) {
+            try {
+            JmxTableHandle tableHandle = requireNonNull(
+                    jmxMetadata.getTableHandle(new SchemaTableName(JmxMetadata.HISTORY_SCHEMA_NAME, tableName)),
+                    format("tableHandle is null for table [%s]", tableName));
+
+                // we are rounding here, so that records from different nodes and different tables will have matching
+                // timestamps (for joining/grouping etc)
+                long dumpTs = roundToPeriod(nowTs);
+
+                List<Object> row = jmxRecordSetProvider.getLiveRow(
+                        tableHandle,
+                        jmxRecordSetProvider.getColumnTypes(tableHandle.getColumns()),
+                        dumpTs);
+                jmxStatsHolder.addRow(tableName, row);
+            }
+            catch (Exception ex) {
```

Comment:
shouldn't we catch `Throwable` here. 
If `Error` is thrown from execution the dumper thread will just die and we will not even see the stack trace.